packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -811,18 +903,53 @@ class MemoryController extends DisposableController
   }
 
   Reference buildTreeFromAllData() {
-    // Nothing to build - no snapshots exists.
-    if (snapshots.isEmpty) return null;
+    topNode ??= LibraryReference(this, libraryRootNode, null);
 
-    if (topNode == null) {
-      topNode = LibraryReference(this, libraryRootNode, null);
+    AllocationsMonitorReference monitorRoot;
+    var anyAnalyses = false;
+    for (final reference in topNode.children) {
+      if (reference is AllocationsMonitorReference) {
+        monitorRoot = reference;
+      }
+      anyAnalyses |= reference is AnalysesReference;
+    }
 
+    if (snapshots.isNotEmpty && !anyAnalyses) {
       // Create Analysis entry.
       final analysesRoot = AnalysesReference();
       analysesRoot.addChild(AnalysisReference(''));
       topNode.addChild(analysesRoot);
     }
 
+    if (monitorAllocations.isNotEmpty) {
+      var createRoot = false;
+      var createChild = false;
+
+      if (monitorRoot != null) {
+        final AllocationMonitorReference monitor = monitorRoot.children.first;
+        if (monitor.dateTime != monitorTimestamp) {
+          monitorRoot.removeLastChild();
```

Comment:
document why you remove the last child even though you compared against the first child.

Reply:
Done.